google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -143,118 +219,94 @@ def create(
                 [import_schema_uri][google.cloud.aiplatform.v1beta1.ImportDataConfig.import_schema_uri],
                 e.g. jsonl file.
-            project: Optional[str]=None,
+            project: Optional[str] = None,
                 Project to upload this model to. Overrides project set in
                 aiplatform.init.
-            location: Optional[str]=None,
+            location: Optional[str] = None,
                 Location to upload this model to. Overrides location set in
                 aiplatform.init.
-            credentials: Optional[auth_credentials.Credentials]=None,
+            credentials: Optional[auth_credentials.Credentials] = None,
                 Custom credentials to use to upload this model. Overrides
                 credentials set in aiplatform.init.
-            sync (bool):
+            request_metadata: Optional[Sequence[Tuple[str, str]]] = ()
+                Strings which should be sent along with the request as metadata.
+            sync: (bool) = True
                 Whether to execute this method synchronously. If False, this method
                 will be executed in concurrent Future and any downstream object will
                 be immediately returned and synced when the Future has completed.
+
         Returns:
             dataset (Dataset):
                 Instantiated representation of the managed dataset resource.
-        """
-        utils.validate_display_name(display_name)
 
-        is_tabular_dataset_metadata = (
-            metadata_schema_uri == schema.dataset.metadata.tabular
-        )
-
-        if bool(bq_source) and not is_tabular_dataset_metadata:
-            raise ValueError(
-                "A tabular metadata_schema uri must be used when using a BigQuery source"
-            )
-
-        api_client = cls._instantiate_client(location=location, credentials=credentials)
+        Raises:
+            ValueError if import_schema_uri is set when creating a tabular dataset.
+        """
 
-        # If this is tabular enrich the dataset metadata with source
-        dataset_metadata = {}
-        if is_tabular_dataset_metadata:
-            if gcs_source:
-                dataset_metadata = {"input_config": {"gcs_source": {"uri": gcs_source}}}
-            elif bq_source:
-                dataset_metadata = {
-                    "input_config": {"bigquery_source": {"uri": bq_source}}
-                }
+        utils.validate_display_name(display_name)
+        cls.metadata_schema_uri = metadata_schema_uri
+        import_data_config = None
+
+        if cls.metadata_schema_uri in [schema.dataset.metadata.tabular]:
+            if import_schema_uri:
+                raise ValueError(
+                    f"{cls.metadata_schema_uri} does not support import_schema_uri."
+                )
+            datasource = TabularDatasource(gcs_source, bq_source)
+        elif cls.metadata_schema_uri in [schema.dataset.metadata.image]:
```

Comment:
We can probably just check for tabular and then assume everything else is non-tabular instead of checking that it matches image, etc.